scripts/release.sh

```
@@ -308,12 +429,94 @@ release_promote() {
+
+  # Nada de artefato velho passando por novo (mesma regra do fluxo Linux):
+  # um .dmg/.zip de versão ou arquitetura anterior em $out viciaria o
+  # `ls` abaixo e subiria binário errado pra release certa.
+  run rm -f "$out"/*.dmg "$out"/*.dmg.sha512 || true
+
+  build_mac "$VERSION" "$out"
+
+  if [ "$DRY" = 0 ]; then
+    # Seleção ANCORADA na versão do build (não apenas "qualquer .dmg"):
+    # `ls *.dmg | head -1` pregaria o primeiro em ordem alfabética.
+    dmg="$(ls "$out"/*"-$VERSION-"*.dmg 2>/dev/null | head -1 || true)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)